server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraUsersRepositoryModule.java

```diff
@@ -45,7 +46,10 @@
     @Override
     public void configure() {
         bind(CassandraUsersRepository.class).in(Scopes.SINGLETON);
-        bind(UsersRepository.class).to(CassandraUsersRepository.class);
+        bind(ReadOnlyUsersLDAPRepository.class).in(Scopes.SINGLETON);
+        bind(CassandraOrLdapUsersRepository.class).in(Scopes.SINGLETON);
+        bind(UsersRepository.class).to(CassandraOrLdapUsersRepository.class);
```

Comment:
It's not obvious to me why we would make configuration files decide which user repository to use and not all other services. What is the rational of making this configuration ?